examples/systemcalls/systemcalls.c

```diff
@@ -6,94 +6,163 @@
-*   to be an absolute path.
-* @param ... - A list of 1 or more arguments after the @param count argument.
-*   The first is always the full path to the command to execute with execv()
-*   The remaining arguments are a list of arguments to pass to the command in execv()
-* @return true if the command @param ... with arguments @param arguments were executed successfully
-*   using the execv() call, false if an error occurred, either in invocation of the
-*   fork, waitpid, or execv() command, or if a non-zero return value was returned
-*   by the command issued in @param arguments with the specified arguments.
-*/
-
+ * @param count -The numbers of variables passed to the function. The variables are command to execute.
+ *   followed by arguments to pass to the command
+ *   Since exec() does not perform path expansion, the command to execute needs
+ *   to be an absolute path.
+ * @param ... - A list of 1 or more arguments after the @param count argument.
+ *   The first is always the full path to the command to execute with execv()
+ *   The remaining arguments are a list of arguments to pass to the command in execv()
+ * @return true if the command @param ... with arguments @param arguments were executed successfully
+ *   using the execv() call, false if an error occurred, either in invocation of the
+ *   fork, waitpid, or execv() command, or if a non-zero return value was returned
+ *   by the command issued in @param arguments with the specified arguments.
+ */
 bool do_exec(int count, ...)
 {
     va_list args;
     va_start(args, count);
-    char * command[count+1];
+    char *command[count + 1];
     int i;
-    for(i=0; i<count; i++)
+    for (i = 0; i < count; i++)
     {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
 
     va_end(args);
 
-    return true;
+    /*
+     * TODO:
+     *   Execute a system command by calling fork, execv(),
+     *   and wait instead of system (see LSP page 161).
+     *   Use the command[0] as the full path to the command to execute
+     *   (first argument to execv), and use the remaining arguments
+     *   as second argument to the execv() command.
+     *
+     */
+    if (command[0][0] != '/') // Command must be an absolute path to binary
+    {
+        return false;
+    }
+
+    pid_t pid = fork();
+    if (pid == -1) // Failure
+    {
+        perror("fork failed!");
+        return false;
+    }
+
+    if (pid == 0) // Child process
+    {
+        execv(command[0], command);
```

Comment:
@Sriram-Y what happens if execv fails, because the path at `command[0]` does not exist?  I suspect your hack to check for `/` is avoiding this failure case which wasn't properly handled.